joomla/joomla-cms
# [4.0] Introduce inlined and lazy attributes to Web Assets

Pull Request for Issue # .

### Summary of Changes
Two more options were added to the assets definition: 
- inline: will inline the source of a js or stylesheet
- lazy: will render a `<link href="path/to/file" rel=print class=js-lazy-loaded data-rel=stylesheet />` but will **not** handle the lazyloding js part (eg switch the rel value to whatever is in the `data-rel` )


### Testing Instructions
Apply the PR and check the admin login and dashboard pages (the source and the DOM)

Lazy load a stylesheet
![Screenshot 2021-03-21 at 16 56 50](https://user-images.githubusercontent.com/3889375/111911617-9bcaf980-8a66-11eb-8406-92fb03f054fe.png)

![Screenshot 2021-03-21 at 16 57 09](https://user-images.githubusercontent.com/3889375/111911620-9cfc2680-8a66-11eb-8573-e287a1a0e5f5.png)

Inlining a js script `"inlined": true` in the joomla.asset.json
![Screenshot 2021-03-21 at 16 57 20](https://user-images.githubusercontent.com/3889375/111911621-9d94bd00-8a66-11eb-8417-b0cf5b119fea.png)

Inlining a stylesheet
`$wa->registerAndUseStyle('f-a-1', 'administrator/templates/atum/css/vendor/fontawesome-free/fontawesome.css', ['inlined' => true], [], []);`

![Screenshot 2021-03-21 at 16 57 36](https://user-images.githubusercontent.com/3889375/111911623-9e2d5380-8a66-11eb-8afa-9c71d1248644.png)



### Documentation Changes Required


# DO NOT MERGE THIS


@Fedik your input here?
